pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -130,12 +132,15 @@ func (cgc *containerGC) removeOldestN(containers []containerGCInfo, toRemove int
 	return containers[:numToKeep]
 }
 
-// removeOldestNSandboxes removes the oldest toRemove sandboxes and returns the resulting slice.
+// removeOldestNSandboxes removes the oldest inactive toRemove sandboxes and
+// returns the resulting slice.
 func (cgc *containerGC) removeOldestNSandboxes(sandboxes []sandboxGCInfo, toRemove int) []sandboxGCInfo {
 	// Remove from oldest to newest (last to first).
 	numToKeep := len(sandboxes) - toRemove
 	for i := numToKeep; i < len(sandboxes); i++ {
-		cgc.removeSandbox(sandboxes[i].id)
+		if !sandboxes[i].active {
+			cgc.removeSandbox(sandboxes[i].id)
```

Comment:
If there is an active sandbox here, returning `sandboxes[:numToKeep]` doesn't make sense to me.

Although we don't use the return value, but the code may confuse people.